Objects/memoryobject.c

```diff
@@ -3303,10 +3305,53 @@ typedef struct {
     const char *it_fmt;
 } memoryiterobject;
 
+#define _PyMemoryViewIter_CAST(PTR) ((memoryiterobject *)(PTR))
+
+static PyObject *
+memoryiter_new(PyObject *self, int reversed)
+{
+    assert(reversed == 0 || reversed == 1);
+
+    if (!PyMemoryView_Check(self)) {
```

Comment:
Can this be called from Python? If not, consider making this an assertion. `memoryview` is not a base type.

Reply:
Called from Python as in, the user can pass something evil to it, but since memory objects can't be subclassed, I don't think that's possible. I guess it's fine to leave it as-is.